api/application.go

```diff
@@ -9,9 +11,36 @@ var EndpointApplications = Endpoint + "applications/"
 
 // https://discord.com/developers/docs/interactions/slash-commands#create-global-application-command-json-params
 type CreateCommandData struct {
-	Name        string                  `json:"name"`
-	Description string                  `json:"description"`
-	Options     []discord.CommandOption `json:"options"`
+	Name                string                  `json:"name"`
+	Description         string                  `json:"description"`
+	Options             []discord.CommandOption `json:"options"`
+	NoDefaultPermission bool                    `json:"default_permission"`
```

Comment:
This should have a `json:"-"` tag so that it doesn't get confused as a valid
JSON field (because it isn't).

The field could be overridden inside a JSON object wrapper:

```go
type rawCreateCommandData CreateCommandData

type createCommandData struct {
	rawCreateCommandData
	DefaultPermission bool `json:"default_permission"`
}
```